### PR TITLE
Fix teleport command

### LINKED ISSUE #1484: TR1X bug: /tp push block doesn't work
Steps to reproduce:

1. `/play vilca`
2. `/tp push block`

The game says it teleported you, but in reality it doesn't do anything.

Same with the object `door` – you need to be specific, and refer to `door 1`, `door 2` etc.

### PR DESCRIPTION
**Pull request by rr-**
#### Checklist

- [x] I have read the [coding conventions](https://github.com/LostArtefacts/TR1X/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description

Resolves #1484 and #1486.

### CODE REVIEW

**Review by aredfan — APPROVED** [review]

File: src/game/lara/lara_cheat.c [hunk 1]
```diff
@@ -368,3 +371,80 @@ bool Lara_Cheat_KillEnemy(const int16_t item_num)
...
+    g_LaraItem->pos.y = y;
+    g_LaraItem->pos.z = z;
+    g_LaraItem->floor = height;
+
+    if (g_LaraItem->room_number != room_num) {
+        const int16_t item_num = g_LaraItem - g_Items;
+        Item_NewRoom(item_num, room_num);
+    }
+
+    if (g_LaraItem->object_number == O_LARA) {
```
Comment (lahm86) [inline src/game/lara/lara_cheat.c]: Would this ever not be true?

**Review by lahm86 — COMMENTED** [review]
Looks good, but it seems to return failure if you tp when beside a pushblock e.g. room 42 in Midas. Not a big issue imo. I suppose this means "you're already there".

image

File: src/game/lara/lara_cheat.c (hunk 1 above)
Comment (rr-) [reply src/game/lara/lara_cheat.c]: True

**Comment by rr-** [comment]
@lahm86 fixed

**Review by lahm86 — APPROVED** [review]

**Pull request merged**